businessRules/ideas.js

```diff
@@ -1,9 +1,73 @@
-const { getAllIdeas } = require('../models/ideas');
+const {
+    getAllIdeas,
+    createIdea,
+    updateIdea,
+    deleteIdea,
+} = require('../models/ideas');
 
 function getIdeas() {
     return getAllIdeas();
 }
 
+function sortedByDonationSumIdeas() {
+    const ideasList = getIdeas();
+
+    const sortedIdeasList = ideasList.sort(
+        (a, b) => a.totalDonationSum - b.totalDonationSum
+    );
+    return sortedIdeasList;
+}
+
+function createIdeas(header, description, askedSum, userId) {
+    if (!header) {
+        throw new Error('Error. Please, check your header.');
+    }
+
+    if (!description) {
+        throw new Error('Error. Please, check your description.');
+    }
+
+    if (!askedSum) {
+        throw new Error('Error. Please, check your asked sum.');
+    }
+
+    if (!userId) {
+        throw new Error('Error. Please, check if you are logged in.');
+    }
+
+    if (header && description && askedSum && userId) {
```

Comment:
This check is now redundant, since you're checking all these parameters in previous `if` statements

Reply:
so, that means, that I can simply leave 
createIdea(header, description, askedSum, userId);
without this part:
"if (header && description && askedSum && userId) "?

Reply:
Updated and tested.